grafast/grafserv/src/servers/whtatwg-node-server/index.ts

```diff
@@ -0,0 +1,115 @@
+import { createServerAdapter } from '@whatwg-node/server'
+
+import { GrafservBase } from "../../core/base.js";
+import type {
+  GrafservConfig,
+  RequestDigest,
+  Result,
+} from "../../interfaces.js";
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace Grafast {
+    interface RequestContext {
+        whatwg: {
+            request: Request
+        }
+    }
+  }
+}
+
+/** @experimental */
+export class WhatwgGrafserv extends GrafservBase {
+  protected whatwgRequestToGrafserv(
+    request: Request
+  ): RequestDigest {
+    const url = new URL(request.url);
+    return {
+      httpVersionMajor: 1,
+      httpVersionMinor: 0,
+      isSecure: url.protocol === 'https://',
+      method:  request.method,
+      path: url.pathname,
+      headers: this.processHeaders(request.headers),
+      getQueryParams() {
+        return Object.fromEntries(url.searchParams.entries()) as Record<string, string>;
+      },
+      async getBody() {
+        const text = await request.text()
+        return {
+          type: "text",
+          text,
+        };
+      },
+      requestContext: {
+        whatwg: {request}
+      },
+      preferJSON: true,
```

Comment:
Disappointing, this opts out of some performance enhancements.

Reply:
I'm honestly not sure about this. How can I tell if this can be set to false?